samcli/lib/remote_invoke/lambda_invoke_executors.py

```diff
@@ -86,41 +94,60 @@ def _execute_action(self, payload: str):
             raise ErrorBotoApiCallException(client_ex) from client_ex
 
     @abstractmethod
-    def _execute_lambda_invoke(self, payload: str):
-        pass
+    def _execute_lambda_invoke(self, payload: str) -> RemoteInvokeIterableResponseType:
+        raise NotImplementedError()
 
 
 class LambdaInvokeExecutor(AbstractLambdaInvokeExecutor):
     """
     Calls "invoke" method of "lambda" service with given input.
     """
 
-    def _execute_lambda_invoke(self, payload: str) -> dict:
+    def _execute_lambda_invoke(self, payload: str) -> RemoteInvokeIterableResponseType:
         LOG.debug(
             "Calling lambda_client.invoke with FunctionName:%s, Payload:%s, parameters:%s",
             self._function_name,
             payload,
             self.request_parameters,
         )
-        return cast(dict, self._lambda_client.invoke(**self.request_parameters))
+        lambda_response = self._execute_boto_call(self._lambda_client.invoke)
+        if self._remote_output_format == RemoteInvokeOutputFormat.RAW:
+            yield RemoteInvokeResponse(lambda_response)
+        if self._remote_output_format == RemoteInvokeOutputFormat.DEFAULT:
+            yield RemoteInvokeResponse(cast(StreamingBody, lambda_response.get(PAYLOAD)).read().decode("utf-8"))
+            log_result = lambda_response.get(LOG_RESULT)
+            if log_result:
+                yield RemoteInvokeLogOutput(base64.b64decode(log_result).decode("utf-8"))
```

Comment:
Feels a little weird to me that these two cases are handled differently here when we've already created different formatters to handle the different output types. I guess I'm wondering how come the outputs themselves are different and not just the format of the outputs?

Reply:
If you mean why a log event is different then the response, is because they are written into different streams. Actual response (aka Payload) is written into `stdout` where log events are printed into `sdterr`, so that customers can pipe them to their other command line utilities. That is why we have two different events to distinguish them so that each can be consumed by a different consumer.

Reply:
My question was more about the output itself. It seems a little strange to me that the output for RAW and DEFAULT is different, it's not just the format that changes between the two output modes.

Reply:
The DEFAULT output format is a more readable output format that prints the returned payload and LogResult if any. The RAW output format prints the full API boto response for users to parse.